dispatch/static/manager/src/js/pages/Widgets/ZoneIndexPage.js

```diff
@@ -7,10 +7,37 @@ import ItemList from '../../components/ItemList'
 
 import * as zonesActions from '../../actions/ZonesActions'
 
+const DEFAULT_LIMIT = 15
+
 class ZoneIndexPageComponent extends React.Component {
 
   componentWillMount() {
-    this.props.listZones(this.props.token)
+    this.props.listZones(this.props.token, this.getQuery())
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.isNewQuery(prevProps, this.props)) {
+      this.props.listZones(this.props.token, this.getQuery())
+    }
+  }
+
+  getQuery() {
+    var query = {
+      limit: DEFAULT_LIMIT,
+    }
+
+    if (this.props.location.query.q) {
+      query.q = this.props.location.query.q
+    }
+    return query
+  }
+
+  isNewQuery(prevProps, props) {
+    return prevProps.location.query.q !== props.location.query.q
+  }
+
+  handleSearchZones(query) {
```

Comment:
This method can be taken out -- you can just call `this.props.searchZones` directly